monkey/monkey_island/cc/ui/src/services/AuthService.js

```diff
@@ -29,16 +33,60 @@ export default class AuthService {
       .then(response => response.json())
       .then(response => {
         if(response.meta.code === 200){
-          this._removeToken();
+          this._removeToken(this.AUTH_TOKEN_NAME_IN_LOCALSTORAGE);
         }
         return response;
       });
   }
 
   authFetch = (url, options) => {
-    return this._authFetch(url, options);
+    return this._authFetch(url, options).then(res => {
+      if(res.status === 401){
+        this._refreshAuthToken();
+        return this._authFetch(url, options);
+      }
+      return res;
+    });
   };
 
+  _refreshAuthToken = () => {
+    let refreshToken = this._getToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE);
+    if (refreshToken) {
+      this._fetchTokenPairFromRefreshToken()
+        .then(response => response.json().then(data => ({status: response.status, body: data})))
+        .then(object => {
+          if(object.status === 200) {
+            this._setTokenPairFromResponseObject(object);
+          } else if (object.status === 401) {
+            this._removeToken(this.AUTH_TOKEN_NAME_IN_LOCALSTORAGE);
+            this._removeToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE);
+          }
+        })
+    }
+  }
+
+  _fetchTokenPairFromRefreshToken() {
+    const options = {
+      method: 'POST',
+      body: JSON.stringify({
+        [this.REFRESH_TOKEN_NAME_IN_RESPONSE]: this._getToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE)
```

Comment:
Why is this in square brackets?

Reply:
I can't find the link but it needs square brackets so it will take the value of the variable when sending the request.

Reply:
@shreyamalviya that's the correct syntax for creating [computed property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names)